### PR TITLE
Add plugin for UI

### PR DESCRIPTION
**Pull request by jni**

### CODE REVIEW

**Comment by jni** [comment]
status update: as of napari 0.4.8rc3, with this PR you can do `napari --with zarpaint -- path/to/file.tiff` and it works for (a) creating/reopening a zarr file and using it as labels, and (b) painting into that file. Chunks are specified as an int tuple that is `ast.literal_eval`d, e.g. `(1, 1, 256, 256)`.

I'll keep pushing a few widgets to this branch, and updating tests. :grimacing: 

CC @GenevieveBuckley

**Comment by jni** [comment]
@tlambert03 I got the dims sorter working :tada: but I have this weird behavior where if I change the values (`dims.order`, `dims.axis_labels`) programmatically, the list updates *only* on hover of the widget. How do I hook up those events to the widget refresh? I couldn't find where in the napari layer list to find that pattern...

If you install the plugin from this branch you can do `napari path/to/image.tif --with zarpaint 'Dims Sorter'` to test. (Then setting the values in the console.)

**Comment by tlambert03** [comment]
I think you might be blocking the events that the Qt view is listening to in your `move_indices` function:
```python
def move_indices(axes_list, order):
    with axes_list.events.blocker_all():
```

however, `SelectableEventedList` already has `move/move_multiple` methods that take care of this for you:

try this:
```python

# instead of this
# dims.events.order.connect(
#     lambda event, axes_list=root: move_indices(axes_list, event.value)
# )

# try this:
dims.events.order.connect(lambda event: root.move_multiple(event.value))

```

**Comment by jni** [comment]
I should have been clearer: yes I block the events because you don't want to update the order when you're updating the list, but adding a `axes_list.events.reordered()` at the end didn't help.

> however, SelectableEventedList already has move/move_multiple methods that take care of this for you:

Of course it does. 😂 It didn't occur to me that moving all of them to 0 with `order` would be the same thing as reordering them. 😂 You don't want to know how long I spent debugging this. 😂 😂 😂 

Having said that, I'm not sure it does exactly what I need, because the order in the list is not `range(ndim)`, which is what `move_multiple(order)` assumes?

File: zarpaint/_dims_chooser.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        root = SelectableEventedList(
+            [AxisModel(dims, i) for i in range(dims.ndim)]
+        )
+        root.events.reordered.connect(
+            lambda event, dims=dims: set_dims_order(dims, event.value)
+        )
+        dims.events.order.connect(
+            lambda event, axes_list=root: move_indices(axes_list, event.value)
+        )
+        view = QtListView(root)
```
Comment (tlambert03) [inline zarpaint/_dims_chooser.py:108]:
```
from qtpy.QtCore import QModelIndex
view.model().dataChanged()  # try this
view.model().dataChanged(QModelIndex(0,0,None),QModelIndex(10,0,None))
```
or

```python
        dims.events.order.connect(self._on_dims_order)

    def 
move_indices(axes_list, event.value)
```

**Comment by GenevieveBuckley** [comment]
:tada:

**Pull request merged**